framework/grid/ActionColumn.php

```diff
@@ -139,7 +150,7 @@ protected function initDefaultButtons()
                     'title' => Yii::t('yii', 'View'),
                     'aria-label' => Yii::t('yii', 'View'),
                     'data-pjax' => '0',
-                ], $this->buttonOptions);
+                ], isset($this->buttonOptions['view']) ? $this->buttonOptions['view'] : $this->buttonOptions);
```

Comment:
This will cause weird stuff to happen if I set some button options but not others.
````
$buttonOptions['weird1'] = ['someOption' => 'someValue']
$buttonOptions['weird2'] = ['someOption' => 'someValue']
````

Will cause the following configuration for the view (and any other default) button:
````
[
    'title' => Yii::t('yii', 'View'),
    'aria-label' => Yii::t('yii', 'View'),
    'data-pjax' => '0',
    'weird1' => [someOption' => 'someValue']
    'weird2' => [['someOption' => 'someValue']
]
````